File: web/src/api/menu.ts
import request, { type ApiResponse, type PageResult, type PageParams } from './request'

export interface Menu {
  id: number
  parent_id: number
  title: string
  path: string
  component: string
  icon: string
  sort: number
  type: number
  status: number
  permission: string
  layout: string
  tenant_id: number
  created_at: string
  updated_at: string
  children?: Menu[]
}

export interface CreateMenuParams {
  parent_id?: number
  title: string
  path: string
  component?: string
  icon?: string
  sort?: number
  type?: number
  status?: number
  permission?: string
  layout?: 'embedded' | 'fullscreen'
}

export interface UpdateMenuParams {
  parent_id?: number
  title?: string
  path?: string
  component?: string
  icon?: string
  sort?: number
  type?: number
  status?: number
  permission?: string
  layout?: 'embedded' | 'fullscreen'
}

export interface MenuListParams extends PageParams {}

export interface AssignRoleMenusParams {
  role_id: number
  menu_ids: number[]
}

const menuApi = {
  createMenu(params: CreateMenuParams): Promise<ApiResponse<Menu>> {
    return request.post('/menus', params)
  },

  getMenuList(params?: MenuListParams): Promise<ApiResponse<PageResult<Menu>>> {
    return request.get('/menus', { params })
  },

  getMenuTree(): Promise<ApiResponse<Menu[]>> {
    return request.get('/menus/tree')
  },

  getMenuByID(menuID: number): Promise<ApiResponse<Menu>> {
    return request.get(`/menus/${menuID}`)
  },

  updateMenu(menuID: number, params: UpdateMenuParams): Promise<ApiResponse<Menu>> {
    return request.put(`/menus/${menuID}`, params)
  },

  deleteMenu(menuID: number): Promise<ApiResponse<null>> {
    return request.delete(`/menus/${menuID}`)
  },

  assignRoleMenus(params: AssignRoleMenusParams): Promise<ApiResponse<null>> {
    return request.post('/menus/assign-role', params)
  },

  getRoleMenus(roleID: number): Promise<ApiResponse<Menu[]>> {
    return request.get(`/menus/role/${roleID}`)
  }
}

export default menuApi

File: backend/internal/router/menu.go
package router

import (
	"fayhub/internal/controller"
	"fayhub/internal/middleware"

	"github.com/gin-gonic/gin"
)

type MenuRouter struct{}

func (s *MenuRouter) Init(router *gin.Engine) {
	menuGroup := router.Group("/api/menus")
	menuGroup.Use(middleware.JwtAuthMiddleware())
	menuGroup.Use(middleware.TenantMiddleware())

	menuGroup.POST("", middleware.SuperAdminMiddleware(), controller.ControllerGroupApp.MenuController.CreateMenu)
	menuGroup.GET("", controller.ControllerGroupApp.MenuController.GetMenuList)
	menuGroup.GET("/tree", controller.ControllerGroupApp.MenuController.GetMenuTree)
	menuGroup.GET("/:menuID", controller.ControllerGroupApp.MenuController.GetMenuByID)
	menuGroup.PUT("/:menuID", middleware.SuperAdminMiddleware(), controller.ControllerGroupApp.MenuController.UpdateMenu)
	menuGroup.DELETE("/:menuID", middleware.SuperAdminMiddleware(), controller.ControllerGroupApp.MenuController.DeleteMenu)

	menuGroup.POST("/assign-role", middleware.SuperAdminMiddleware(), controller.ControllerGroupApp.MenuController.AssignRoleMenus)
	menuGroup.GET("/role/:roleID", controller.ControllerGroupApp.MenuController.GetRoleMenus)
}
